ui/panels-plugin/src/plugins/gauge-chart/GaugeChartOptionsEditor.tsx

```diff
@@ -11,13 +11,44 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { useEffect, useRef, useState } from 'react';
+import { TextField } from '@mui/material';
 import { OptionsEditorProps } from '@perses-dev/plugin-system';
-import { Box } from '@mui/material';
 import { GaugeChartOptions } from './gauge-chart-model';
 
 export type GaugeChartOptionsEditorProps = OptionsEditorProps<GaugeChartOptions>;
 
 export function GaugeChartOptionsEditor(props: GaugeChartOptionsEditorProps) {
-  const { value } = props;
-  return <Box>{JSON.stringify(value)}</Box>;
+  const ref = useRef<HTMLInputElement>(null);
```

Comment:
Rather than copy and paste, should we just create a JSONEditor component than add it to each Plugin. Similar to this PR:

https://github.com/perses/perses/pull/627/files#diff-e98abc89e67f736bd08bda2b6e011f4e555adcde880f8c5ac17b743d987411d7R18

https://github.com/perses/perses/pull/627/files#diff-95552669c95c924f6a5299409039403a40ba51baa6fe2bb1ae6580a851ea17afR57

Reply:
Yeah definitely, just added it to the components package, I was being lazy before since this shouldn't be around very long once I get to adding actual form controls